src/routes.js

```diff
@@ -68,14 +73,21 @@ const utxoForAddresses = (dbApi: DbApi, { logger, apiConfig }: ServerConfig) =>
 ) => {
   validateAddressesReq(apiConfig.addressesRequestLimit, req.body)
   logger.debug('[utxoForAddresses] request is valid')
-  const result = await dbApi.utxoForAddresses(req.body.addresses)
+  const utxos: Array<UtxoForAddressesDbResult> = await dbApi.utxoForAddresses(req.body.addresses)
+  const utxosWithTokens = await appendTokens(dbApi, utxos)
   logger.debug('[utxoForAddresses] result calculated')
-  return (result.map(row => (
+  return (utxosWithTokens.map(row => (
     {
-      utxo_id: `${row.tx_hash}${row.tx_index}`,
       ...row,
+      utxo_id: `${row.tx_hash}${row.tx_index}`,
+      amount: getCoinObject(
+        row.amount,
+        row.tokens,
+      ),
+      txOutDbId: undefined,
+      tokens: undefined,
```

Comment:
doesn't this mean it just returns no tokens all the time?

Reply:
It's removing them from the top object (which came from the DB), but they're encapsulated in the CoinObject tthree lines above

Reply:
I'd add at least a comment clarifying what's going on as at first sight, it looks pretty frightening